examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+"""
+Title: Document Images Restoration/Enhancer.
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/01/18
+Last modified: 2022/01/21
+Description: Example of Document Image Restoration coupled with Tesseract OCR engine to improve text readability.
+"""
+"""
+## Introduction
+
+This example explains a "Image Enhancer/Denoiser" using UNet architecture. The problem we
+focus here is enhancing the document images which are deteriorated through external
+degradations like blur, corrupt text blocks etc... ,we have followed up on few seminal
+papers as reference and used simple pretrained UNet as encoder and Efficient Sub-Pixel
+CNN as decoder. we do not attempt super-resolution as primary task hence we return output
+image in the same dimension as input. we compute WER (Word Error Rate) metric to evaluate
+model performance helping us to assess the text information gain in real world scenarios
+when coupled with tesseract OCR engine.
+
+**References:**
+- [Enhancing OCR Accuracy with Super
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+- [Improving the Perceptual Quality of Document
+Images Using Deep Neural
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+from tensorflow.keras import callbacks
+from tensorflow.keras.applications import resnet_v2
+
+"""
+## Dataset
+we have created a synthetic document dataset that depicts several real world document
```

Comment:
'we', here W should be uppercase. There is some similar typo across the file, please address those too.